metals/src/main/scala/scala/meta/internal/metals/Compilations.scala

```diff
@@ -88,19 +90,17 @@ final class Compilations(
   ): CancelableFuture[b.CompileResult] = {
     val params = new b.CompileParams(targets.asJava)
     targets.foreach(target => isCompiling(target) = true)
-    val compilation = connection.compile(params)
-    val task = for {
-      result <- compilation.asScala
-      _ <- {
-        lastCompile = isCompiling.keySet
+    val compilation = connection
+      .compile(params)
+      .whenComplete((_, error) => {
```

Comment:
Let's try to use Scala's `Future` operations over `CompletableFuture`. It's easy to mess up with the execution context in the Java API in my experience and it's not idiomatic to for example guard against `error == null`